### ISSUE TITLE
Barbarians - still spawning in colony.

### ISSUE THREAD

**Issue by bob0bob** [ldtteam/minecolonies#1657]
Barbarians spawns still in the middle of the colony on average.  ABout 20 - 30 blocks away from town hall.  I've not had one spawn even close to the colony limit.
I have it at 200.

I have the default barbarians set to 5.   They appear to be easy.    I have 13 citizens.  2 guards.  I have many buildings,  even decor stuff build.  Market place, library, park.  

I've not seen the barbarians ability grow.  The size of the horde does but the difficulty is still set to beginner.

FYI,
 Last spawn,  they spawned under a apple tree and where killed by the tree. 

PS.  My town has killer apple trees.

![2017-11-18_19 45 43](https://user-images.githubusercontent.com/30982580/32986531-a1fe1b36-cc99-11e7-87e7-a0b706fabbc4.png)
 (that is what I call it).

**Comment by bob0bob** [comment]
minecolonies-universal-
minecraft 1.12.2-
minecolonies 0.8.5203


Barbarians are still spawning within 20 - 30 blocks of town hall.   Not even close to spawning outside of town.

Which makes building walls pointless,  since they just spawn on top of citizens and such. 

This has been the case for about 2 months now.

**Comment by Raycoms** [comment]
@Asherslab do you think you could improve that a bit?

**Comment by Asherslab** [comment]
i can most certainly try to improve it. bob0bob - the problem is that we cannot spawn barbarians in unloaded chunks, and as such we need to spawn them as close to the colony as there are loaded chunks. The best thing i can think of, is possibly finding the outermost hut in the colony and then creating a "circle" (as we were before) from that hut (Obv testing that the hut is in a loaded chunk) around the colony. Obviously i haven't tested that and the workings of it, but it ***Could*** work

**Comment by Raycoms** [comment]
The current code already checks for loaded chunks in a certain distance. We
just have to investigate and trigger the numbers a bit to improve it.

2017-11-27 8:14 GMT+00:00 Asherslab <notifications@github.com>:

**Comment by bob0bob** [comment]
I hope I'm doing nothing wrong.  I just reporting what I see in the builds, even if they are alpha builds and you might be aware of them or working on them and it not complete.   Just wanted to make sure I reported what I saw in case your guys didn't notice it.  With know it could be duplicate issues that are already know.  I try to do a search to see if I locate something already reported, but that doesn't mean you know about them without an issue.

As far as chunk loading.   I'm not a minecraft expert and do not completely understand that. Is there a way to determine how many chunks are loaded.  I have the draw distant set to 20.   I have chunk updates set to 3 in  settings.    I know,  I have 2 lumberjacks and they are on each corner if the current size of the colony.  They have a distance of 130+ blocks between the hut block and they keep working all the time.   I've not notice anyone stopping because they chunk got unloaded.    But again,  maybe you handle that already.

I've even been 200 blocks away from town center and barbarians spawned and attacked the citizens before I could get back.  The guards didn't do anything.  The barbarians spawn at dusk, before citizens are in their huts so it is a slaughter fest for them. 

But let me know what to try and I'll do it and let you know the results.  If you ever need a video, I can use replay mod to capture them if that helps.
